python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -388,6 +399,43 @@ def terminate_scheduled_nodes(self):
         self.nodes_to_terminate = []
         self.update_worker_list()
 
+    def drain_nodes_via_gcs(self):
+        node_ips = [
+            self.provider.internal_ip(node_id)
+            for node_id in self.nodes_to_terminate
+        ]
+        raylet_ids_to_drain = {
+            self.load_metrics.raylet_id_by_ip[ip]
+            for ip in node_ips
+        }
+
+        logger.info(f"Draining raylets with ids {raylet_ids_to_drain}.")
+        request = gcs_service_pb2.DrainNodeRequest(drain_node_data=[
+            gcs_service_pb2.DrainNodeData(node_id=raylet_id)
+            for raylet_id in raylet_ids_to_drain
+        ])
+        try:
+            response = self.gcs_node_info_stub.DrainNode(request)
+            drained_raylet_ids = {
+                status_item.node_id
+                for status_item in response.drain_node_status
+            }
```

Comment:
Do we have an RPC timeout here? Otherwise this could hang for a long time if the node is not responsive.

Reply:
Right, timeouts are important..
 
@rkooo567 5 sec is probably good enough since there isn't anything heavy going on server end before response?

Reply:
I think 5 seconds are probably too short for the rpc timeout? What about 30s? Is it a blocking call?

Reply:
30 it is

Reply:
For this one I would go with something shorter like at most 5s. The reason is stalling the autoscaler is critical, much worse than producing a wrong error message.

Reply:
Makes sense.

Reply:
+1 to 5 seconds. I think we should track the exceptions here in the prometheus metrics to see if it causes any issues. This is critical because autoscaler updates every 5 seconds.
@DmitriGekhtman , do you mind including the exceptions in promtheus?

Reply:
Great idea -- drain node exception counter added to prometheus metrics.